### PR TITLE
FEATURE: Add scheduled job to purge expired ignored users

### PR DESCRIPTION
**Pull request by khalilovcmded**
## Why?

This is part of the [Ability to ignore a user feature](https://meta.discourse.org/t/ability-to-ignore-a-user/110254/8).

> The maximum ignore duration will be 4 months, so let’s go ahead and set up a task to clear these ignores after 4 months, as we have to assume all beta ignores are “max duration”.

### CODE REVIEW

File: spec/jobs/purge_expired_ignored_users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+require_dependency 'jobs/scheduled/purge_expired_ignored_users'
+
+describe Jobs::PurgeExpiredIgnoredUsers do
+  subject { Jobs::PurgeExpiredIgnoredUsers.new.execute({}) }
+
+  context "with no ignored users" do
+    it "does nothing" do
+      subject
```
Comment (ZogStriP) [inline spec/jobs/purge_expired_ignored_users_spec.rb]: Do we need that line?

File: spec/jobs/purge_expired_ignored_users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    let(:john) { Fabricate(:user, username: "john") }
+
+    before do
+      Fabricate(:ignored_user, user: tarek, ignored_user: matt)
+      Fabricate(:ignored_user, user: tarek, ignored_user: john)
+    end
+
+    context "when no expired ignored users" do
+      it "does nothing" do
+        subject
```
Comment (ZogStriP) [inline spec/jobs/purge_expired_ignored_users_spec.rb]: Ditto ^^

File: spec/jobs/purge_expired_ignored_users_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    end
+
+    context "when there are expired ignored users" do
+      it "purges expired ignored users" do
+        freeze_time(5.months.ago) do
+          fred = Fabricate(:user, username: "fred")
+          Fabricate(:ignored_user, user: tarek, ignored_user: fred)
+        end
+
+        expect { subject }.to change { IgnoredUser.count }.from(3).to(2)
```
Comment (ZogStriP) [inline spec/jobs/purge_expired_ignored_users_spec.rb]: I would also make sure we can't find the old "IgnoreUser" record anymore.

Comment (khalilovcmded) [reply spec/jobs/purge_expired_ignored_users_spec.rb]: Yes, makes sense. I will actually remove this check, and just check we can't find that specific IgnoredUser anymore.

**Pull request merged**